spatialos-sdk/src/worker/connection.rs

```diff
@@ -487,6 +495,30 @@ impl Connection for WorkerConnection {
         }
     }
 
+    fn send_component_update_serialized(
```

Comment:
That's not correct. The C API will only call the vtable serialize method iff `schema_type` is `nullptr`. The function which calls the vtable serialize function (inside the C API) looks roughly like:

```
OutgoingData ManagedClientHandle::Serialize() && {
  auto* serialize = get_serialize_function(vtable)
  if (serialize && !schema_type) {
    SchemaDataType* serialized_schema_type = nullptr;
    serialize(component_id, vtable.user_data, user_handle.get(), &serialized_schema_type);
    ...
  }
  ...
}
```

so this shouldn't segfault in that case.

However, it _would_ segfault if the user provides null for both `schema_type` and `user_handle` (in the `Worker_ComponentUpdate`).

Reply:
Awesome :smile:

Reply:
Yup yup! I've started to add methods like this, where the user can specifically choose to opt out of vtable serialization. This doesn't fully address the needs of users who want to avoid vtable serialization, but it's a first step in that direction.